### PR TITLE
active link

### PR DESCRIPTION
**Pull request by drowsy810301**
need rerun python install.py

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,8 @@ migrations/
 
 # Config files
 nthuoj.ini
+nthucsoj.py
+nthucsoj.pyc
```
Comment (nickelcheng) [inline .gitignore]: *.pyc is included in ```*.py[cod]```, so you don't have to write this.

File: install.py [hunk 2]
```diff
@@ -37,6 +37,10 @@
     pwd = getpass.getpass()
     write_ini_file(host, db, user, pwd)
 
+if not os.path.isfile('nthucsoj.py'):
```
Comment (nickelcheng) [inline install.py]: I prefer using ```emailPwd.py``` as file name.

File: install.py [hunk 3]
```diff
@@ -37,6 +37,10 @@
...
+    # Setting nthucsoj.py
+    email_host_pwd = getpass.getpass('nthucsoj@gmail.com password : ')
```
Comment (nickelcheng) [inline install.py]: I think we can customize email address for users, too.

File: users/models.py [hunk 4]
```diff
@@ -132,3 +132,13 @@ class Notification(models.Model):
 
     def __unicode__(self):
         return str(self.id)
+
+class UserProfile(models.Model):
+    user = models.OneToOneField(User)
+    activation_key = models.CharField(max_length=40, blank=True)    
+      
+    def __str__(self):
```
Comment (nickelcheng) [inline users/models.py]: Please use ```def __unicode__(self)``` for python 2.

File: users/urls.py [hunk 5]
```diff
@@ -33,4 +33,6 @@
     url(r'^list/$', views.list, name='list'),
     url(r'^submit/$', views.submit, name='submit'),
     url(r'^profile/(?P<username>\w+)$', views.profile, name='profile'),
+    url(r'^confirm/(?P<activation_key>\w+)/', ('users.views.register_confirm'), name="confirm"),
```
Comment (nickelcheng) [inline users/urls.py]: Please use `'`

**Comment by nickelcheng** [comment]
Please give an alert when a inactive user login.

**Comment by drowsy810301** [comment]
fix all

File: users/urls.py [hunk 6]
```diff
@@ -33,4 +33,6 @@
     url(r'^list/$', views.list, name='list'),
     url(r'^submit/$', views.submit, name='submit'),
     url(r'^profile/(?P<username>\w+)$', views.profile, name='profile'),
+    url(r'^confirm/(?P<activation_key>\w+)/', views.register_confirm, name="confirm"),
```
Comment (nickelcheng) [inline users/urls.py]: Please use ```name='confirm'```

File: users/views.py [hunk 7]
```diff
@@ -118,10 +123,36 @@ def profile(request, username):
...
+            user=User.objects.get(username=username)
+
+            # Create and save user profile
+            new_profile = UserProfile(user=user, activation_key=activation_key)
+            new_profile.save()
+
+            #Send email with activation key
+            email_subject = 'Account confirmation'
+            email_body = "Hey %s, thanks for signing up.\n \
+To activate your account, click the link below.\n" % (username) + \
```
Comment (nickelcheng) [inline users/views.py]: Please take care of indent.

**Comment by nickelcheng** [comment]
Auto logout after a registration complete.

**Comment by drowsy810301** [comment]
OK,I think I fix all!

File: users/views.py [hunk 8]
```diff
@@ -169,3 +200,25 @@ def user_login(request):
...
+    if request.user.is_authenticated():
+        HttpResponseRedirect(reverse('index:index'))
+
+    '''check if there is UserProfile which matches 
+    the activation key (if not then display 404)
+    '''
+    user_profile = get_object_or_404(UserProfile, activation_key=activation_key)
+    user = user_profile.user
+    user.is_active = True
+    user.save()
```
Comment (nickelcheng) [inline users/views.py]: Please write a info log for modifying user active status.

**Comment by nickelcheng** [comment]
I'm ok to this PR after log added. @CHIAHSINHOU please test this.

**Comment by drowsy810301** [comment]
The security of email host user should be like the following image.
961149_839831389397076_196426894_n

**Comment by drowsy810301** [comment]
Logging added.

**Comment by nickelcheng** [comment]
Please mention about email security setting in readme

**Comment by drowsy810301** [comment]
Mention email security setting in readme

File: users/views.py [hunk 9]
```diff
@@ -118,13 +123,39 @@ def profile(request, username):
...
+            new_profile.save()
+
+            #Send email with activation key
+            email_subject = 'Account confirmation'
+            email_body = 'Hey %s, thanks for signing up.\n ' % (username) + \
+            'To activate your account, click the link below.\n'  + \
+            request.META['HTTP_HOST'] + \
+            reverse('users:confirm', kwargs={'activation_key': activation_key})
+
+            send_mail(email_subject, email_body, 'nthucsoj@gmail.com', [email], fail_silently=False)
```
Comment (hydai) [inline users/views.py]: 關於寄信功能有兩個建議：
1. 是否能用 Thread 把他隔開執行，這樣可以避免程序阻塞在此，有等待寄信的延遲感。
2. 應該用 try...catch 把 send_mail 包住，處理所有例外情況，避免寄信失敗造成 500 錯誤等

**Comment by drowsy810301** [comment]
Wrap send email func in try and catch.Please review again

File: users/views.py [hunk 10]
```diff
@@ -118,13 +123,45 @@ def profile(request, username):
...
+
+            #Send email with activation key
+            email_subject = 'Account confirmation'
+            email_body = 'Hey %s, thanks for signing up.\n ' % (username) + \
+            'To activate your account, click the link below.\n'  + \
+            request.META['HTTP_HOST'] + \
+            reverse('users:confirm', kwargs={'activation_key': activation_key})
+
+            try:
+                send_mail(email_subject, email_body, 'nthucsoj@gmail.com', [email], fail_silently=False)
```
Comment (henryyang42) [inline users/views.py]: 如@hydai 建議的
send_mail可以用thread包起來
不然跳轉頁面要等很久

**Comment by drowsy810301** [comment]
Fixed.Use Thread to send email.

File: index/templates/index/index.html [hunk 11]
```diff
@@ -33,6 +33,13 @@
 
 {% block body_block %}
   <!-- clock -->
+  <script>
+      if ('{{ alert_info }}' == 'mailbox'){
+          message = 'Please go to your mailbox and click the confirmation ' + 
+          'link;otherwise , your account would not be activated!';
+          alert(message);
+      }
+  </script>
```
Comment (bruce3557) [inline index/templates/index/index.html]: Prefer to make this into a function and call it in js file.
Let HTML code separates javascript code as possible.

Comment (drowsy810301) [inline index/templates/index/index.html]: But it cannot catch template tag in js file.

Comment (nickelcheng) [inline index/templates/index/index.html]: How about pass parameters

**Comment by drowsy810301** [comment]
OK,I fixed it.

**Comment by drowsy810301** [comment]
Does this PR have any problem?

**Comment by nickelcheng** [comment]
@CHIAHSINHOU do you test this?

**Comment by bruce3557** [comment]
Fine with this PR, waiting for testing

**Comment by CHIAHSINHOU** [comment]
I've tested this PR. Go to merge.

**Pull request merged**